analysis_configs/sequences_s_channel_scouting.py

```diff
@@ -150,6 +150,40 @@ def add_analysis_branches(events):
     
     return events
 
+def add_dark_quark_matching(events):
+    #performs delta R matching between AK8 jets and dark quarks and saves the indices of the matched jets in a new branch
+    #(uses new functions for now, should check if it also works with the old ones)
+
+    jet_eta = events.FatJet_eta
+    jet_phi = events.FatJet_phi
+
+    dark_quark_eta = events.MatrixElementGenParticle_eta
+    dark_quark_phi = events.MatrixElementGenParticle_phi
+
+    # matching with the first dark quark
+    dR1 = event_vars.delta_r_dark_quark(dark_quark_eta[:,0], dark_quark_phi[:,0], jet_eta[:,:], jet_phi[:,:]) <= 0.8
+    dR1 = ak.values_astype(dR1, int)
+    matched_index1 = ak.argmax(dR1, axis=1)
+    matched_index1 = ak.where(ak.any(dR1, axis=1), matched_index1, -1)
+
+    #matching with the second dark quark 
+    dR2 = event_vars.delta_r_dark_quark(dark_quark_eta[:,1], dark_quark_phi[:,1], jet_eta[:,:], jet_phi[:,:]) <= 0.8
+    dR2 = ak.values_astype(dR2, int)
+    matched_index2 = ak.argmax(dR2, axis=1)
+    matched_index2 = ak.where(ak.any(dR2, axis=1), matched_index2, -1)
+
+    #create a combined list of matched indices
+    matched_indices = ak.concatenate([matched_index1[:, None], matched_index2[:, None]], axis=1)
+    matched_indices = ak.sort(matched_indices, axis=1)
+    matched_indices = matched_indices[matched_indices >= 0]
+
+    #add new branch to the events 
+    events["FatJet_matchedIdx"] = matched_indices
```

Comment:
I am not sure I understand the code: is `matched_indices` of the same length as the FatJet collection? Given the `argmax` and the `concatenate` I would tend to say this has a different size.
If same size, keep it like this, if different size, then it will be better to call it differently (not starting with `FatJet_`) so that it is not considered in the FatJet collection.

Reply:
it is not the same size, it basically contains the indices for the matched jets, meaning 2 entries in most cases, even if the event has 3 FatJets for example
when I run it it actually gives me a warning that it is considered a standalone because it has a different size compared to most branches in the FatJet collection
I think because of this, it also creates an nFatJet_matchedIdx branch
I could also rename it, but I'm not sure if that would make it more confusing. Since it's used on the FatJets as a sort of mask I wanted to name it FatJet, but if you have other name suggestions let me know

Reply:
This is different, this warning comes from the fact that ak cannot infer the type of the array, and places a question mark.
And this comes from the fact that the delta phi min could be undefined from a computing point of view (e.g. there is no jet in an event), but this never happens in practice because we require events to have jets!
So there is nothing you can do, it is automatically handled :)